### DISCUSSION TITLE
Unmount issues/Trying to understand semantics of volumeHandle

### DISCUSSION THREAD

**Discussion by RustyF** [kubernetes-csi/csi-driver-smb#879] [Q&A]
Context: EKS 1.30, using recent driver version, multiple pods, each with their own PVC and PVs mounting the same remote share.

Hi there, we're still having odd problems where pods either aren't terminating and some mounts become unreachable. Looking at the logs, I can see we get a lot of these unmount errors, "The device mount path ..... is still mounted by other references". I'm thinking it's down to where we have unrelated pods using the same remote share. 

Could someone clarify the semantics of volumeHandle?  Should the volume handle be unique per remote share OR unique per pod per remote share?

E.g. is this right, deployment A and B both using the same remote share (and on the same node).

Deployment A, pod A, PVC A, PV A connecting to share foo.com/share/dir => volumeHandle = "foo.com/share#dir#A"
Deployment B, pod B, PVC B, PV B connecting to share foo.com/share/dir => volumeHandle  = "foo.com/share#dir#B"

Thanks in advance

**Comment by andyzhangx** [comment]
the error is related to https://github.com/kubernetes-csi/csi-driver-smb/issues/353#issuecomment-926518175, you should define a unique volumeHandle for each PV(two PVs could share the same smb server address and sharename), while the volumeHandle setting is not related to your error.

**Reply by RustyF** [reply]
Thanks @andyzhangx, it looks like my example reflects this; given PVC/PVs are unique to each deployment, the volumehandle is effectively unique to each deployment (A and B in my example).

Regarding the comment, I did see this but I didn't understand the outcome. Is this highlighting a bug or an unavoidable design fact? Is there anything I can do to avoid this?
